libs/mockito-extensions/src/main/java/org/codice/ddf/test/mockito/StackContainsDoPrivilegedCalls.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.test.mockito;
+
+import java.security.AccessController;
+import java.util.Arrays;
+import org.hamcrest.CustomMatcher;
+import org.mockito.Incubating;
+
+/**
+ * Hamcrest {@link org.hamcrest.CustomMatcher} that can be used to determine a stack contains a
+ * certain number of {@code AccessController.doPrivileged()} calls.
```

Comment:
❓ Should we add some candidate circumstances when testing for this should occur? And when it shouldn't? If we abuse this we might end up in trouble. Should every privileged call have tests for this stuff? Or only critical security manager paths?

Reply:
Great questions. At this point I was thinking of presenting this to other developers, see what they think and what the right/wrong usages are, and then provide some general guidelines based on that.